### PR TITLE
Fix double counting users

### LINKED ISSUE #3717: ATI Analytics idclient field
**Is your feature request related to a problem? Please describe.**
We are currently double counting users between Simorgh and Orbit-based pages.  We are able to fix some of that duplication.

**Describe the solution you'd like**
If the user has an `atuserid` cookie set on their device for the current domain (bbc.co.uk OR bbc.com), then:
- Read the cookie value - which will be a json string
- Extract the value of the `val` attribute (which will be a uuid)
- Add this to all ATI calls as an additional parameter `idclient=${val}` - we should add this as the second query string parameter (after `s`)

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**
Note that this will not fix _all_ duplicate users in our data, but is a good first step

### PR DESCRIPTION
**Pull request by samora**
Resolves #3717 

**Overall change:** We are currently double counting users between Simorgh and Orbit-based pages. This fixes some of that duplication.

**Code changes:**

- Adds `idclient` query param to ATI calls. The value is read from `atuserid` cookie.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [x] This PR requires manual testing

### CODE REVIEW

File: src/app/lib/analyticsUtils/index.js [hunk 1]
```diff
@@ -125,6 +125,19 @@ export const getReferrer = (platform, origin, previousPath) => {
   return null;
 };
 
+export const getAtUserId = platform => {
+  if (platform === 'amp') {
+    return null;
+  }
+
+  if (onClient()) {
+    const atuserid = Cookie.getJSON('atuserid');
```
Comment (dr3) [inline src/app/lib/analyticsUtils/index.js:130]: To anyone else reviewing whos curious, `Cookie.getJSON` is fail safe :D https://github.com/js-cookie/js-cookie/blob/155e16b95aa39afc9e8ddbc202a065afee5866b5/src/js.cookie.mjs#L96-L98

File: src/app/lib/analyticsUtils/index.js [hunk 2]
```diff
@@ -125,6 +125,19 @@ export const getReferrer = (platform, origin, previousPath) => {
   return null;
 };
 
+export const getAtUserId = platform => {
+  if (platform === 'amp') {
+    return null;
+  }
+
+  if (onClient()) {
```
Comment (dr3) [inline src/app/lib/analyticsUtils/index.js:129]: Can combine these `if`s

```suggestion
  if (onClient() && platform !== 'amp') {
```
Technically, `onClient()` is never true on AMP pages, as we dont run client JS, but nice to be explicty

File: src/app/lib/analyticsUtils/index.js [hunk 3]
```diff
@@ -125,6 +125,19 @@ export const getReferrer = (platform, origin, previousPath) => {
...
+    const atuserid = Cookie.getJSON('atuserid');
+    return (atuserid && atuserid.val) || null;
```
Comment (dr3) [inline src/app/lib/analyticsUtils/index.js]:
```suggestion
    return pathOr(null, ['val'], atuserid)
```

**Review by dr3 — CHANGES_REQUESTED** [review]

File: src/app/lib/analyticsUtils/index.js [hunk 4]
```diff
@@ -125,6 +125,19 @@ export const getReferrer = (platform, origin, previousPath) => {
   return null;
 };
 
+export const getAtUserId = platform => {
+  if (platform === 'amp') {
```
Comment (dr3) [inline src/app/lib/analyticsUtils/index.js]: Can remove this `if` now :)

File: src/app/lib/analyticsUtils/index.test.js [hunk 5]
```diff
@@ -387,3 +388,23 @@ describe('getPublishedDatetime', () => {
     expect(publishedTime).toEqual(null);
   });
 });
+
+describe('getAtUserId', () => {
+  returnsNullWhenOffClient(getAtUserId);
```
Comment (sareh) [inline src/app/lib/analyticsUtils/index.test.js:393]: I think this is missing:
```suggestion
  returnsNullWhenOffClient(getAtUserId());
```

**Review by dr3 — APPROVED** [review]

**Review by sareh — APPROVED** [review]
👍 Looks great. Prior to merging, it'd be good for testers to find relevant assets and to ensure the `idclient` query param is being sent in the ATI call on Canonical pages only.

**Pull request merged**